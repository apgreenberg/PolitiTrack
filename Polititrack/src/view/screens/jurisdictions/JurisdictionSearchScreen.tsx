import React from 'react';
import { StyleSheet, Button, View, SafeAreaView } from 'react-native';
import { Picker } from '@react-native-picker/picker';
import 'react-native-gesture-handler';

import { addContentToDatabase } from '../../../model/database/handleContent';
import { UserInfo, UserContent } from '../../../model/database/userInterfaces';

import { selectCurrentUser, useAppSelector } from '../../../redux/hooks';

import { states, state, stateIdMap } from '../../../assets/states';

/**
 * Displays the jurisdiction search screen and adds the user's choice to the database
 * @returns - the functional components representing the screen
 */
const JurisdictionSearchScreen: React.FC = () => {
  const currentUser: UserInfo = useAppSelector(selectCurrentUser);
  const [selectedState, setSelectedState] = React.useState('Alabama');

  return (
    <SafeAreaView>
      <View style={styles.pageContainer}>
        <Picker
          selectedValue={selectedState}
          onValueChange={(itemValue: string) => setSelectedState(itemValue)}
        >
          {states.map((state: string, key: number) => (
            <Picker.Item label={state} value={state} key={key}></Picker.Item>
          ))}
        </Picker>
      </View>
      <View>
        <Button
          title={'Add ' + selectedState}
          onPress={() =>
            addContentToDatabase(currentUser, 'jurisdictions', {
              id: stateIdMap[selectedState as state],
              name: selectedState,
              lastUpdated: new Date().getTime(),
            })
          }
        />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  pageContainer: {
    height: '80%',
    flexDirection: 'column',
    justifyContent: 'space-between',
  },
});
export default JurisdictionSearchScreen;
