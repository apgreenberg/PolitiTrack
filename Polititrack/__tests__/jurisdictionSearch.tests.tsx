import React from 'react';
import TestRenderer from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import JurisdictionSearchScreen from '../src/view/screens/jurisdictions/JurisdictionSearchScreen';
import { useSelectorMock, userData } from './common';

/**
 * Tests the Jurisdiction Search Screen
 */
describe('Jurisdiction search tests', () => {
  test('Viewing the jurisdiction selection page', async () => {
    useSelectorMock.mockReturnValue(userData);
    const component = TestRenderer.create(
      <MockedProvider mocks={[]} addTypename={false}>
        <JurisdictionSearchScreen />
      </MockedProvider>
    );
    const myJurisdictionsScreen = component.toJSON();
    expect(myJurisdictionsScreen).toMatchSnapshot();
  });
});
