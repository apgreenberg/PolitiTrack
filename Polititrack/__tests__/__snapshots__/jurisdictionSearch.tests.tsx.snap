// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jurisdiction search tests Viewing the jurisdiction selection page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "flexDirection": "column",
        "height": "80%",
        "justifyContent": "space-between",
      }
    }
  >
    <View>
      <RNCPicker
        items={
          Array [
            Object {
              "label": "Alabama",
              "testID": undefined,
              "textColor": undefined,
              "value": "Alabama",
            },
            Object {
              "label": "Alaska",
              "testID": undefined,
              "textColor": undefined,
              "value": "Alaska",
            },
            Object {
              "label": "Arizona",
              "testID": undefined,
              "textColor": undefined,
              "value": "Arizona",
            },
            Object {
              "label": "Arkansas",
              "testID": undefined,
              "textColor": undefined,
              "value": "Arkansas",
            },
            Object {
              "label": "California",
              "testID": undefined,
              "textColor": undefined,
              "value": "California",
            },
            Object {
              "label": "Colorado",
              "testID": undefined,
              "textColor": undefined,
              "value": "Colorado",
            },
            Object {
              "label": "Connecticut",
              "testID": undefined,
              "textColor": undefined,
              "value": "Connecticut",
            },
            Object {
              "label": "Delaware",
              "testID": undefined,
              "textColor": undefined,
              "value": "Delaware",
            },
            Object {
              "label": "Florida",
              "testID": undefined,
              "textColor": undefined,
              "value": "Florida",
            },
            Object {
              "label": "Georgia",
              "testID": undefined,
              "textColor": undefined,
              "value": "Georgia",
            },
            Object {
              "label": "Hawaii",
              "testID": undefined,
              "textColor": undefined,
              "value": "Hawaii",
            },
            Object {
              "label": "Idaho",
              "testID": undefined,
              "textColor": undefined,
              "value": "Idaho",
            },
            Object {
              "label": "Illinois",
              "testID": undefined,
              "textColor": undefined,
              "value": "Illinois",
            },
            Object {
              "label": "Indiana",
              "testID": undefined,
              "textColor": undefined,
              "value": "Indiana",
            },
            Object {
              "label": "Iowa",
              "testID": undefined,
              "textColor": undefined,
              "value": "Iowa",
            },
            Object {
              "label": "Kansas",
              "testID": undefined,
              "textColor": undefined,
              "value": "Kansas",
            },
            Object {
              "label": "Kentucky",
              "testID": undefined,
              "textColor": undefined,
              "value": "Kentucky",
            },
            Object {
              "label": "Louisiana",
              "testID": undefined,
              "textColor": undefined,
              "value": "Louisiana",
            },
            Object {
              "label": "Maine",
              "testID": undefined,
              "textColor": undefined,
              "value": "Maine",
            },
            Object {
              "label": "Maryland",
              "testID": undefined,
              "textColor": undefined,
              "value": "Maryland",
            },
            Object {
              "label": "Massachusetts",
              "testID": undefined,
              "textColor": undefined,
              "value": "Massachusetts",
            },
            Object {
              "label": "Michigan",
              "testID": undefined,
              "textColor": undefined,
              "value": "Michigan",
            },
            Object {
              "label": "Minnesota",
              "testID": undefined,
              "textColor": undefined,
              "value": "Minnesota",
            },
            Object {
              "label": "Mississippi",
              "testID": undefined,
              "textColor": undefined,
              "value": "Mississippi",
            },
            Object {
              "label": "Missouri",
              "testID": undefined,
              "textColor": undefined,
              "value": "Missouri",
            },
            Object {
              "label": "Montana",
              "testID": undefined,
              "textColor": undefined,
              "value": "Montana",
            },
            Object {
              "label": "Nebraska",
              "testID": undefined,
              "textColor": undefined,
              "value": "Nebraska",
            },
            Object {
              "label": "Nevada",
              "testID": undefined,
              "textColor": undefined,
              "value": "Nevada",
            },
            Object {
              "label": "New Hampshire",
              "testID": undefined,
              "textColor": undefined,
              "value": "New Hampshire",
            },
            Object {
              "label": "New Jersey",
              "testID": undefined,
              "textColor": undefined,
              "value": "New Jersey",
            },
            Object {
              "label": "New Mexico",
              "testID": undefined,
              "textColor": undefined,
              "value": "New Mexico",
            },
            Object {
              "label": "New York",
              "testID": undefined,
              "textColor": undefined,
              "value": "New York",
            },
            Object {
              "label": "North Carolina",
              "testID": undefined,
              "textColor": undefined,
              "value": "North Carolina",
            },
            Object {
              "label": "North Dakota",
              "testID": undefined,
              "textColor": undefined,
              "value": "North Dakota",
            },
            Object {
              "label": "Ohio",
              "testID": undefined,
              "textColor": undefined,
              "value": "Ohio",
            },
            Object {
              "label": "Oklahoma",
              "testID": undefined,
              "textColor": undefined,
              "value": "Oklahoma",
            },
            Object {
              "label": "Oregon",
              "testID": undefined,
              "textColor": undefined,
              "value": "Oregon",
            },
            Object {
              "label": "Pennsylvania",
              "testID": undefined,
              "textColor": undefined,
              "value": "Pennsylvania",
            },
            Object {
              "label": "Puerto Rico",
              "testID": undefined,
              "textColor": undefined,
              "value": "Puerto Rico",
            },
            Object {
              "label": "Rhode Island",
              "testID": undefined,
              "textColor": undefined,
              "value": "Rhode Island",
            },
            Object {
              "label": "South Carolina",
              "testID": undefined,
              "textColor": undefined,
              "value": "South Carolina",
            },
            Object {
              "label": "South Dakota",
              "testID": undefined,
              "textColor": undefined,
              "value": "South Dakota",
            },
            Object {
              "label": "Tennessee",
              "testID": undefined,
              "textColor": undefined,
              "value": "Tennessee",
            },
            Object {
              "label": "Texas",
              "testID": undefined,
              "textColor": undefined,
              "value": "Texas",
            },
            Object {
              "label": "Utah",
              "testID": undefined,
              "textColor": undefined,
              "value": "Utah",
            },
            Object {
              "label": "Vermont",
              "testID": undefined,
              "textColor": undefined,
              "value": "Vermont",
            },
            Object {
              "label": "Virginia",
              "testID": undefined,
              "textColor": undefined,
              "value": "Virginia",
            },
            Object {
              "label": "Washington",
              "testID": undefined,
              "textColor": undefined,
              "value": "Washington",
            },
            Object {
              "label": "West Virginia",
              "testID": undefined,
              "textColor": undefined,
              "value": "West Virginia",
            },
            Object {
              "label": "Wisconsin",
              "testID": undefined,
              "textColor": undefined,
              "value": "Wisconsin",
            },
            Object {
              "label": "Wyoming",
              "testID": undefined,
              "textColor": undefined,
              "value": "Wyoming",
            },
          ]
        }
        onChange={[Function]}
        selectedIndex={0}
        style={
          Array [
            Object {
              "height": 216,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View>
    <View
      accessibilityRole="button"
      accessibilityState={Object {}}
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Add Alabama
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
